src/components/products-list/product-card.tsx

```diff
@@ -10,10 +19,85 @@ export interface ProductCardProps {
   price: string;
   state: string;
   image_url: string;
+  isFavorite: boolean;
+  refetch: () => void;
 }
 
+export type TitleAndFavoriteProps = {
+  id: number;
+  name: string;
+  isFavorite: boolean;
+  refetch: () => void;
+};
+
+const TitleAndFavorite = ({
+  id,
+  name,
+  isFavorite,
+  refetch,
+}: TitleAndFavoriteProps) => {
+  const { mutate: addFavorite } = useAddFavorite({
+    onSuccess: () => {
+      refetch();
+    },
+    onError: () => {
+      showErrorMessage('Something went wrong');
+    },
+  });
+  const { mutate: removeFavorite } = useRemoveFavorite({
+    onSuccess: () => {
+      refetch();
+    },
+    onError: () => {
+      showErrorMessage('Something went wrong');
+    },
+  });
```

Comment:
I feel this is too much responsability to the component to carry. Components should be mostly visual components and the logic should live in the parent